### PR TITLE
Bug/1444 err msg duplicate attr

### LINKED ISSUE #1444: [BUG]  In create request is possible to create an entity with  N equal attributes
#### Create request:
```
 http://qa-orion-fe-01:1026/v2/entities
Content-Type: application/json
Fiware-Service: test_service_path
Fiware-ServicePath:/test
{"type":"house", "temperature": {"type": "celcius", "value": "34"}, "temperature": {"type": "celcius", "value": "49"},"id": "room7"}
```

#### Create response:
```
Status 201 Created 
```

#### Doc in mongo:
```
{ "_id" : { "id" : "room7", "type" : "house", "servicePath" : "/test" }, "attrNames" : [ "temperature", "temperature" ], "attrs" : { "temperature" : { "type" : "celcius", "creDate" : 1446122837, "modDate" : 1446122837, "value" : "34" }, "temperature" : { "type" : "celcius", "creDate" : 1446122837, "modDate" : 1446122837, "value" : "34" } }, "creDate" : 1446122837, "modDate" : 1446122837 }
```

### PR DESCRIPTION
**Pull request by crbrox**
Fixes #1444 

@fgalan @kzangeli

### CODE REVIEW

File: test/functionalTest/cases/1444_err_msg_duplicate_attr/create_entity_dup_attr.test [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+--REGEXPECT--
+01. POST /v2/entities, creating first entity with id E1
+==========================================================
+HTTP/1.1 422 Unprocessable Entity
+Content-Length: 88
+Content-Type: application/json
+Date: REGEX(.*)
+
+{
+    "description": "duplicated attribute name and id [attr,]",
```
Comment (kzangeli) [inline test/functionalTest/cases/1444_err_msg_duplicate_attr/create_entity_dup_attr.test]: Not really part of this PR, but this extra comma ... would be really nice to have it removed.
New issue?

**Comment by kzangeli** [comment]
LGTM

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 2]
```diff
@@ -2767,7 +2767,7 @@ static bool contextElementPreconditionsCheck
       {
         ContextAttribute* ca = new ContextAttribute(ceP->contextAttributeVector.get(ix));
         buildGeneralErrorResponse(ceP, ca, responseP, SccInvalidModification,
-                                  "duplicated attribute name and id [" + name + "," + id + "]");
+                                  "duplicated attribute /" + name + "/");
         LM_W(("Bad Input (duplicated attribute name: name=<%s> id=<%s>)", name.c_str(), id.c_str()));
         return false; // Error already in responseP
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: I just realized a thing ...

    ContextAttribute* ca = new ContextAttribute( ... );
    ...
    return false;

This must be a leak, right?
An 'old' one but not SO old ...

Easy fix;

    ContextAttribute* ca = ceP->contextAttributeVector.get(ix);

Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Actually it might be an old leak, if we haven't ever had a functest covering this

Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Discussed on skype. Seems that buildGeneralErrorResponse adds it to some vector and that it is later released.
If not so, valgrind tests will tell us.

NTC

**Comment by fgalan** [comment]
LGTM
